Feature/setup ci (#7)
open-rmf/rmf_demos: rmf_demos_panel/README.md

## RMF Demos Panel
This will describes more about the details of `rmf_demos_panel`. In short, this package uses a simple web server
to expose the essential rmf topics/services to the user as web endpoints. Currently, this lite implementation is 
useful for task submission and observe on-going tasks/robots in RMF.

### RMF Dependencies
 - `rmf_task_ros2`
 - `rmf_fleet_msgs`
 - `rmf_demos_dashboard_resources`

### Setup `rmf_demos_panel`

```bash

> python3 -m pip install flask-socketio

Reviewer: ```suggestion
python3 -m pip install Flask flask-socketio flask-cors
```

Reviewer: Or are these installed by `rosdep`?

Author: Good catch. Unfortunately, we cant download `flask-socketio` (only) via `rosdep` due to https://github.com/osrf/rmf_demos/issues/202

npm install --prefix src/rmf/rmf_demos/rmf_demos_panel/rmf_demos_panel/static/
npm run build --prefix src/rmf/rmf_demos/rmf_demos_panel/rmf_demos_panel/static/

colcon build --packages-select rmf_demos_panel
```

### API Endpoints

Endpoints | Type | Parameters | Description
--- | --- | --- | ---
/submit_task | POST | task description json | This supports submission of task. This response with an assigned task_id is the task is accepted
/cancel_task | POST | task_id string | Cancel an existing task in rmf.
/task_list | GET | NA | Get list of task status in RMF (include active and terminated tasks)
/robot_list | GET | NA | Get list of Robot states in RMF
/task_status | SocketIO | NA | Constant broadcast of task list
/robot_states | SocketIO | NA | Constant broadcast of robot list

#### Simple CURL Test

```bash
# Submit Task (POST)
curl -X POST http://localhost:8080/submit_task \
-d '{"task_type":"Loop", "start_time":0, "description": {"start_name": "coe", "finish_name": "pantry", "num_loops":1}}' \
-H "Content-Type: application/json" 

# Get Task List (GET)
curl http://localhost:8080/task_list
```

---

## Run Sample Tasks

**Submit a list of tasks***

> On the right side column of the web UI, users are able to select a file which consists of a list of  

Reviewer: ```suggestion
On the right side column of the panel, users are able to select a file which consists of a list of  
```

1. `gui_server` (port `5000`): Providing the static gui to the web client. Non RMF dependent
2. `api_server` (port `8080`): Hosting all endpoints for gui clients to interact with RMF

> To create your own customize GUI, you will only require to replace the current `gui_server` package with your 

Reviewer: ```suggestion
To create your own custom GUI, you will only require to replace the current `gui_server` package with your 
```

> own custom implementation, while still interact with the existing `api_server`.

Reviewer: ```suggestion
own custom implementation. The `api_server` remains the same.
```

---

## Note
- Edit the `dashboard_config.json` to configure the input of the Demo World GUI Task Submission.

> The dashboard config file is located here: `rmf_dashboard_resources/$WORLD/dashboard_config.json`.

Reviewer: ```suggestion
The dashboard config file is located here: `rmf_demos_dashboard_resources/$WORLD/dashboard_config.json`.
```
